File: practica/patrones-de-diseño/observer/biblioteca/Libro.java
/**
 * Auto Generated Java Class.
 */
public class Libro {
  private String titulo;
  private boolean estaDisponible;
  
  public Libro(String titulo, boolean estadoDisponible){
    this.titulo = titulo;
    this.estaDisponible = estadoDisponible;
  }
  
  public String getTitulo(){
    return titulo;
  }
  
  public void setEstaDisponible(boolean nuevoEstado){
    estaDisponible = nuevoEstado;
  }
  
  public boolean getEstaDisponible(){
    return estaDisponible;
  }
}

File: practica/patrones-de-diseño/typescript/observer/biblioteca/src/Libro.ts
export abstract class Libro{
    protected titulo?:string

    public getTitulo(){
        return this.titulo
    }
}

export class LibroNull extends Libro{
    public getTitulo() {
        return "No existe este libro"
    }
}

export class LibroComun extends Libro{
    constructor(titulo: string) {
        super()
        this.titulo = titulo
    }
}
